### PR TITLE
Display req and limits

### PR DESCRIPTION
**Pull request by hidalgopl**
Added few things to forecast tab:
Screenshot from 2020-11-19 15-35-01

### CODE REVIEW

File: cost_calculator/templates/cost_summary.html [hunk 1]
```diff
@@ -42,7 +52,8 @@
     </thead>
   </table>
   <p>
-    <strong>Note:</strong> Nodeless cost is projected based on cpu and memory resource requests and limits for the applications running on the cluster. Projected cost uses on-demand instance price by default. Spot projections coming soon!
+    <strong>Note:</strong> Nodeless cost is projected based on cpu and memory resource requests and limits for the applications running on the cluster. Projected cost uses on-demand instance price by default. If row is yellow, it means that resource requirements are not defined for given pod.
+      This may falsify calculations and it's consider bad practice. Please set those and come back to get more accurate forecast!
```
Comment (jrroman) [inline cost_calculator/templates/cost_summary.html:56]: nit: some small grammar things
- consider -> considered
- please set those and come back to get more -> please set resource requirements for those pods to get a more

**Review by jrroman — APPROVED** [review]
Just the one comment relating to the wording of the warning. Otherwise LGTM :+1:

File: cost_calculator/app.py [hunk 2]
```diff
@@ -22,50 +23,81 @@
...
 
-def k8s_container_resource_requirements(container):
+def k8s_container_resource_requirements(container) -> Dict[str, int]:
+    """
+    returns dict in format:
+     {"req_cpu": 1, "req_mem": 1, "lim_cpu": 2, "lim_mem": 2}
+    """
     try:
         max_cpu = 0
         max_memory = 0
```
Comment (ldx) [inline cost_calculator/app.py:33]: It's a bit misleading to call these `max_cpu` and `max_memory`, since they are only used as constant zero values it seems.

File: cost_calculator/app.py [hunk 3]
```diff
@@ -22,50 +23,81 @@
...
-            max_memory = max(memory, max_memory)
-    sum_cpu = 0
-    sum_memory = 0
+            resources = k8s_container_resource_requirements(container)
+            req_cpu, req_mem = resources['req_cpu'], resources['req_mem']
+            lim_cpu, lim_mem = resources['lim_cpu'], resources['lim_mem']
+            max_req_cpu = max(req_cpu, max_req_cpu)
+            max_lim_cpu = max(lim_cpu, max_lim_cpu)
+            max_req_memory = max(req_mem, max_req_memory)
+            max_lim_memory = max(lim_mem, max_lim_memory)
```
Comment (ldx) [inline cost_calculator/app.py:82]: In `k8s_container_resource_requirements()` it is already ensured that all keys have a zero value if the container does not have the corresponding resource set, so maybe eliminate doing it here again?

File: cost_calculator/templates/cost_summary.html [hunk 4]
```diff
@@ -42,7 +52,8 @@
     </thead>
   </table>
   <p>
-    <strong>Note:</strong> Nodeless cost is projected based on cpu and memory resource requests and limits for the applications running on the cluster. Projected cost uses on-demand instance price by default. Spot projections coming soon!
+    <strong>Note:</strong> Nodeless cost is projected based on cpu and memory resource requests and limits for the applications running on the cluster. Projected cost uses on-demand instance price by default. If row is yellow, it means that resource requirements are not defined for given pod.
+      This may falsify calculations and it's considered bad practice. Please set resource requirements for those pods to get a more giaccurate forecast!
```
Comment (ldx) [inline cost_calculator/templates/cost_summary.html:56]: I'd rephrase this line like this:

> This may result in inaccurate results in the calculator and not recommended for production systems.

There's also a typo "giaccurate" -> "accurate".

File: cost_calculator/test_instance_selector.py [hunk 5]
```diff
@@ -6,21 +6,92 @@
...
+        for case in cases:
+            got = k8s_pod_resource_requirements(case['pod'])
+            self.assertEqual(got, case['expected'])
+
+
 class TestInstanceSelector(unittest.TestCase):
     def assert_matches(self, cpu, memory, gpu, expected):
         inst_type, _ = self.instance_selector.get_cheapest_instance(cpu, memory, gpu)
         msg = f'{self.instance_selector.cloud}: {cpu}, {memory}, {gpu} != expected {expected}, got {inst_type}'
+        print(msg)
```
Comment (ldx) [inline cost_calculator/test_instance_selector.py:94]: Debug print()?

**Review by ldx — APPROVED** [review]
LGTM. I dropped a few comments, for a couple of stylistic changes.

File: cost_calculator/test_instance_selector.py (hunk 5 above)
Comment (hidalgopl) [reply cost_calculator/test_instance_selector.py]: nice catch ;) yeap.

File: cost_calculator/templates/cost_summary.html (hunk 4 above)
Comment (hidalgopl) [reply cost_calculator/templates/cost_summary.html]: thanks!

File: cost_calculator/app.py (hunk 3 above)
Comment (hidalgopl) [reply cost_calculator/app.py]: good point, I tried to follow previous approach on too many levels;)

File: cost_calculator/app.py (hunk 2 above)
Comment (hidalgopl) [reply cost_calculator/app.py]: removed `max_` prefix in few places :+1:
